File: scripts/demo_topic_subscriber.py
#! /bin/python3

import rospy
from std_msgs.msg import Int32

def callback(msg):
    rospy.loginfo('Recieved [%s]' % msg.data)

def demo_topic_subscriber():
    rospy.init_node('demo_topic_subscriber', anonymous=True)
    rospy.Subscriber('numbers', Int32, callback)

    # Spinning the node
    rospy.spin()

if __name__ == '__main__':
    try:
        demo_topic_subscriber()
    except rospy.ROSInterruptException:
        pass

File: src/demo_topic_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

/* Callback of the topic /numbers */
void number_callback(const std_msgs::Int32::ConstPtr& msg)
{
    ROS_INFO("Recieved  [%d]",msg->data);
}

int main(int argc, char **argv)
{

    /* Initializing ROS node with a name of demo_topic_subscriber */
    ros::init(argc, argv,"demo_topic_subscriber");
    /* Created a nodehandle object */
    ros::NodeHandle nh;
    /* Create a publisher object */
    ros::Subscriber number_subscriber = nh.subscribe("/numbers", 10, number_callback);
    /* Spinning the node */
    ros::spin();
    
    return 0;
}

